utils/src/main/kotlin/Extensions.kt

```diff
@@ -261,6 +261,10 @@ fun String.percentEncode(): String =
         // "+" instead of "%20", so apply the proper mapping here afterwards ("+" in the original string is
         // encoded as "%2B").
         .replace("+", "%20")
+        // "*" is a reserved character in RFC 3986.
+        .replace("*", "%2A")
+        // "~" is an unreserved character in RFC 3986.
+        .replace("%7E", "~")
```

Comment:
Do you think this could lead to issues with `String.fileSystemEncode()` for edge cases where some name starts with "~/" and the tilde is interpreted as home directory?

Reply:
Hmm, actually I don't think so. Tilde-expansion is done by the shell, and we don't do any shell calls.

Reply:
Ok, just wanted to make sure.